From 5a117f55ba0898ec6557cd890c2808ee972077de Mon Sep 17 00:00:00 2001
From: Zhou Peng-B04994 <eagle.zhou@nxp.com>
Date: Mon, 23 Oct 2017 10:58:55 +0800
Subject: [PATCH 2675/5242] MLK-16671-2 - [i.MX8QXP/Malone]: Add vpu malone
 decoder driver

commit  62b076e9093ba5effad7b2f0f4ce0d84bb129429 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add vpu module in device tree and makefile

Signed-off-by: Zhou Peng-B04994 <eagle.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   28 +++++++++++++++++++++---
 drivers/mxc/Kconfig                            |    4 ++++
 drivers/mxc/Makefile                           |    1 +
 3 files changed, 30 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 8859ddb..c7bdef8 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -626,9 +626,13 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			pd_vpu_core: vpu_core {
-				name = "vpu_core";
-				reg = <SC_R_VPUCORE>;
+			pd_vpu_enc: VPU_ENC {
+				reg = <SC_R_VPU_ENC_0>;
+				#power-domain-cells = <0>;
+				power-domains =<&pd_vpu>;
+			};
+			pd_vpu_dec: VPU_DEC {
+				reg = <SC_R_VPU_DEC_0>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_vpu>;
 			};
@@ -2104,6 +2108,24 @@
 		status = "disabled";
 	};
 
+	vpu: vpu@2c000000 {
+		compatible = "nxp,imx8qm-vpu", "nxp,imx8qxp-vpu";
+		reg = <0x0 0x2c000000 0x0 0x1000000>;
+		reg-names = "vpu_regs";
+		interrupts = <0 464 0x4>, /* encoder irq */
+		<0 465 0x4>, /* encoder fiq */
+		<0 466 0x4>, /* decoder irq */
+		<0 467 0x4>, /* decoder fiq */
+		<0 468 0x4>; /* decoder sif */
+		interrupt-names = "enc_irq", "enc_fiq", "dec_irq", "dec_fiq", "dec_sif";
+		clocks = <&clk IMX8QXP_VPU_DEC_CLK>;
+		clock-names = "vpu_clk";
+		assigned-clocks = <&clk IMX8QXP_VPU_DEC_CLK>;
+		assigned-clock-rates = <600000000>;
+		power-domains = <&pd_vpu_dec>;
+		status = "disabled";
+	};
+
 	imx_rpmsg: imx_rpmsg {
 		compatible = "fsl,rpmsg-bus", "simple-bus";
 		#address-cells = <2>;
diff --git a/drivers/mxc/Kconfig b/drivers/mxc/Kconfig
index 72030cf..2d656e4 100755
--- a/drivers/mxc/Kconfig
+++ b/drivers/mxc/Kconfig
@@ -9,6 +9,10 @@ source "drivers/mxc/hantro/Kconfig"
 source "drivers/mxc/mlb/Kconfig"
 source "drivers/mxc/hdp/Kconfig"
 
+if ARCH_MXC_ARM64
+source "drivers/mxc/vpu-malone/Kconfig"
+endif
+
 if ARCH_MXC
 config MXC_IPU
 	bool "Image Processing Unit Driver"
diff --git a/drivers/mxc/Makefile b/drivers/mxc/Makefile
index 4a6644f..632f465 100755
--- a/drivers/mxc/Makefile
+++ b/drivers/mxc/Makefile
@@ -5,4 +5,5 @@ obj-$(CONFIG_MXC_IPU_V3) += ipu3/
 obj-$(CONFIG_MXC_HDMI_CEC) += hdmi-cec/
 obj-$(CONFIG_MXC_MIPI_CSI2) += mipi/
 obj-$(CONFIG_MXC_HANTRO) += hantro/
+obj-$(CONFIG_MXC_VPU_MALONE) += vpu-malone/
 obj-$(CONFIG_MX8_HDP)	+= hdp/
-- 
1.7.9.5

